src/FormButton/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Omit } from '@stoplight/types';
+import _noop = require('lodash/noop');
+import * as React from 'react';
+// TODO: should probably use ajv and json schema
+import * as yup from 'yup';
+
+import { Button, IButtonProps } from '../index';
+
+/**
+ * FORM BUTTON
+ * button that is disabled if the schema doesn't pass validation
+ * TODO, add tooltip to explain why its disabled?
+ */
+interface IFormButton extends Omit<IButtonProps, 'type' | 'onClick'> {
+  schema?: yup.Schema<any>;
+  data?: any;
+  onClick?: (data: any) => void;
+}
+
+const FormButton: React.FunctionComponent<IFormButton> = ({
+  schema,
+  data,
+  loading,
+  disabled,
+  onClick = _noop,
+  ...buttonProps
+}) => {
+  const [isValid, setIsValid] = React.useState<boolean>(false);
+
+  async function validate() {
+    if (!schema) return;
+    try {
+      await schema.validateSync(data);
+    } catch (e) {
+      setIsValid(false);
+    }
+  }
+
+  React.useEffect(() => {
+    setIsValid(true);
+    validate();
```

Comment:
Is there a benefit in defining `validate` as a function outside of the useEffect scope? I believe defining outside will create a new function on each render and defining inside useEffect will memoize it with data and schema 🤔 

```ts
React.useEffect(async () => {
    setIsValid(true);

    if (!schema) return;

    try {
      await schema.validateSync(data);
    } catch (e) {
      setIsValid(false);
    }
}, [data, schema]);
```

Reply:
useEffect, doesn't allow async functions as the input, but I can move the validate function definitions inside of the effect

Reply:
Ah yeah that makes sense because async functions always return a promise but useEffect expects a cleanup function. Yeah you could define it inside would work too!

Reply:
Or just use it as a promise lol

Reply:
```ts
React.useEffect(() => {
    setIsValid(true);

    if (!schema) return;

    schema.validateSync(data).catch(() => setIsValid(false));
}, [data, schema]);
```

Reply:
tried the catch and its not actually catching the error

Reply:
Extracted to a hook